src/services/followMainFeedBroad.js

```diff
@@ -1,41 +1,39 @@
-const stream = require("getstream");
-const { unFollowFeedProcessQueue } = require("../config");
+const stream = require('getstream');
+const {unFollowFeedProcessQueue} = require('../config');
 
-const followMainFeedBroad = async (userId, userIds) => {
-  console.log("Broad Feed => ", userIds)
-  if (!userIds || userIds.length == 0) {
+const followMainFeedBroad = async (userId, userIds, user_anon = false) => {
+  if (!userIds || userIds.length === 0) {
     return;
   }
-
+  const targetFeed = user_anon ? 'user_anon' : 'user';
   const cs = stream.connect(process.env.API_KEY, process.env.SECRET);
 
   const payload = userIds.map((ui) => {
     return {
       source: `main_feed_broad:${userId}`,
-      target: `user:${ui}`,
+      target: `${targetFeed}:${ui}`
     };
   });
-  return await cs.followMany(payload);
+  await cs.followMany(payload);
 };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `followMainFeedBroad` function has been updated to include a new parameter `user_anon`, which defaults to `false`. This is used to determine the target feed. The function now also checks if the `userIds` array is empty before proceeding, which is a good practice to avoid unnecessary operations. However, it's important to note that the function no longer returns the result of `cs.followMany(payload)`. If this return value is not needed elsewhere in your code, then this change is fine. Otherwise, you should add a return statement.

```diff
-  return await cs.followMany(payload);
+  await cs.followMany(payload);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->